ironfist/src/cpp/game/save.cpp

```diff
@@ -527,18 +524,16 @@ int game::SaveGame(char *saveFile, int autosave, signed char baseGame) {
 
 	const xml_schema::base64_binary datbin(dat, sz);
 
-	ofstream os(v9);
+	std::ofstream os(v9);
 
 	ironfist_map::map_t m(datbin);
 
 	for(int i = 0; i < ELEMENTS_IN(this->heroes); i++) {
 		m.hero().push_back(WriteHeroXML(&this->heroes[i]));
 	}
 
-	if (GetScriptContents() != NULL) {
-		m.script(GetScriptContents());
-		SaveMapVariables(m);
-	}
+	m.script(GetScriptContents());
+	SaveMapVariables(m);

```

Comment:
I remembered why I did this!

It's because GetScriptContents() now returns "" if there is none. Lack of a script is just an empty script.

However, I've confirmed that SaveMapVariables is now responsible for the crash.